src/core/CMakeLists.txt

```diff
@@ -1000,6 +1009,8 @@ INCLUDE_DIRECTORIES(
   metadata
   pal
   processing
+  providers
```

Comment:
So apparently memory provider can be used for custom python providers (though I can't really follow how exactly), that makes sense - although I'm not completely sold yet to the idea of (ab)using memory provider for this.

What's the advantage of integrating gdal/ogr in core?